hugegraph-llm/src/hugegraph_llm/operators/llm_op/info_extract.py

```diff
@@ -0,0 +1,273 @@
+    chunked_data = split_string(string, 500)
+    combined_chunks = []
+    current_chunk = ""
+    for chunk in chunked_data:
+        if (
+            llm.num_tokens_from_string(current_chunk) + llm.num_tokens_from_string(chunk)
+            < allowed_tokens
+        ):
+            current_chunk += chunk
+        else:
+            combined_chunks.append(current_chunk)
+            current_chunk = chunk
+    combined_chunks.append(current_chunk)
+
+    return combined_chunks
+
+
+def get_spo_from_result(result):
+    res = []
+    for row in result:
+        row = row.replace("\\n", "").replace("\\", "")
+        pattern = r'\("(.*?)", "(.*?)", "(.*?)"\)'
+        res += re.findall(pattern, row)
+    return res
+
+
+def get_nodes_and_relationships_from_result(result):
+    regex = (
+        r"Nodes:\s+(.*?)\s?\s?Relationships:\s+(.*?)\s?\s?NodesSchemas:\s+(.*?)\s?\s?\s?"
+        r"RelationshipsSchemas:\s?\s?(.*)"
+    )
+    internal_regex = r"\[(.*?)\]"
+    nodes = []
+    relationships = []
+    nodes_schemas = []
+    relationships_schemas = []
+    for row in result:
+        row = row.replace("\n", "")
+        parsing = re.search(regex, row, flags=re.S)
+        if parsing is None:
+            continue
+        raw_nodes = str(parsing.group(1))
+        raw_relationships = parsing.group(2)
+        raw_nodes_schemas = parsing.group(3)
+        raw_relationships_schemas = parsing.group(4)
+        nodes.extend(re.findall(internal_regex, raw_nodes))
+        relationships.extend(re.findall(internal_regex, raw_relationships))
+        nodes_schemas.extend(re.findall(internal_regex, raw_nodes_schemas))
+        relationships_schemas.extend(re.findall(internal_regex, raw_relationships_schemas))
+    result = {
+        "nodes": [],
+        "relationships": [],
+        "nodes_schemas": [],
+        "relationships_schemas": [],
+    }
+    result["nodes"].extend(nodes_text_to_list_of_dict(nodes))
+    result["relationships"].extend(relationships_text_to_list_of_dict(relationships))
+    result["nodes_schemas"].extend(nodes_schemas_text_to_list_of_dict(nodes_schemas))
+    result["relationships_schemas"].extend(
+        relationships_schemas_text_to_list_of_dict(relationships_schemas)
+    )
+    return result
+
+
+class InfoExtract:
+    def __init__(
+        self,
+        llm: BaseLLM,
+        text: str,
+        nodes_schemas=None,
+        relationships_schemas=None,
+        spo=False,
+    ) -> None:
+        self.llm = llm
+        self.text = text
+        self.nodes_schemas = nodes_schemas
+        self.relationships_schemas = relationships_schemas
+        self.spo = spo
+
+    def process(self, chunk):
+        if self.llm.get_llm_type() == "openai":
+            messages = [
+                {"role": "system", "content": self.generate_system_message()},
+                {"role": "user", "content": self.generate_prompt(chunk)},
+            ]
+        else:
```

Comment:
Would it be better to explicitly specify the model type here, since other models will be accessed later?